lib/experimental.ts

```diff
@@ -121,3 +121,7 @@ export function fullImageSavePath(imageSavePath: string | undefined): string {
 
   return path.join(imagePath, uuidv4());
 }
+
+function isTrue(value: boolean | string): boolean {
+  return String(value).toLowerCase() === "true";
```

Comment:
is this actually the right check? the default of this flag it true (I think?), so what happen if the user does not pass this flag?

Reply:
here is the behaviour before this commit:
| flag | value |
|-----|------|
|        | false - do not run app vuln scan |
| --app-vulns | `true` - run app vuln scan |
| --app-vulns=false | `'false'` - run app vuln scan |
| --app-vulns=true | `'true'` - run app vulns scan |

the `--app-vulns=false` flag was being misinterpreted as a true value

Reply:
here is the behaviour after this commit:
| flag | value |
|-----|------|
|        | false - do not run app vuln scan |
| --app-vulns | `true` - run app vuln scan |
| --app-vulns=false | `false` - do not run app vuln scan |
| --app-vulns=true | `true` - run app vulns scan |

Reply:
could you please add a test that verifies the above? take an image, see that the scan returns app vulns, then pass false and see that it doesn't

Reply:
just to confirm what is being asked, a test for `String(value).toLowerCase() === "true"` and moved to it own file/test combo?
or
a test for `experimental.ts` in this file?
https://github.com/snyk/snyk-docker-plugin/blob/feat/app-vulns-parameter/test/lib/experimental.test.ts

Reply:
I've updated the original comment with the screenshots requested

Reply:
for prosperities sake, here are the screenshots for the current production version `1.404.0`

`snyk test --docker --experimental mladkau/snykin`
![Screenshot 2020-09-29 at 17 38 15](https://user-images.githubusercontent.com/70949530/94587723-e5d2df80-027a-11eb-92ed-6a16ec4f7577.png)

`snyk test --docker --experimental --app-vulns=false mladkau/snykin`
![Screenshot 2020-09-29 at 17 38 45](https://user-images.githubusercontent.com/70949530/94587724-e7040c80-027a-11eb-846d-a9930ace886e.png)

`snyk test --docker --experimental --app-vulns mladkau/snykin`
![Screenshot 2020-09-29 at 17 39 09](https://user-images.githubusercontent.com/70949530/94587726-e8353980-027a-11eb-86a4-b2dca25d4d03.png)

`snyk test --docker --experimental --app-vulns=true mladkau/snykin`
![Screenshot 2020-09-29 at 17 39 35](https://user-images.githubusercontent.com/70949530/94587728-e8cdd000-027a-11eb-9960-09e7467aac45.png)

Reply:
you claim that before this PR when you scanned image with `--app-vulns=false` it still returned the app vulns. but after this PR it should not happen any more.
let's add a test that verifies that - a test that would fail without the code in this pr

Reply:
After reviewing some of the existing tests, the most appropriate course of action I've identified to fulfil the testing requirement, is to create 3 new system tests `(--app-vulns | --app-vulns=false | --app-vulns=true)` in [./test/system/experimental.test.ts](https://github.com/snyk/snyk-docker-plugin/blob/feat/app-vulns-parameter/test/system/experimental.test.ts) that follows the style of [./test/system/application-scans.test.ts](https://github.com/snyk/snyk-docker-plugin/blob/feat/app-vulns-parameter/test/system/application-scans.test.ts) in order to cover the gap where there are no `--app-vulns` related code being run under the `--experimental` flag.

@shaimendel shall I proceed with this approach?

Reply:
I think that testing `--app-vulns` and `--app-vulns=false` should be enough :) thank you!